### PR TITLE
add rules for constant returning zero and one function

### PR DESCRIPTION
**Pull request by oxinabox**
Is this right?
Should I have some type constraints somewhere 
or does `@scalar_rule` give me that for free?

### CODE REVIEW

**Comment by willtebbutt** [comment]
LGTM, not sure about the type constraints though.

**Comment by oxinabox** [comment]
@willtebbutt  can you approve if you approve?

File: test/rulesets/Base/base.jl [hunk 1]
```diff
@@ -103,5 +103,22 @@ end
         rrule_test(identity, randn(rng), (randn(rng), randn(rng)))
         rrule_test(identity, randn(rng, 4), (randn(rng, 4), randn(rng, 4)))
     end
+
+    @testset "Constants" begin
+        function test_constant(f, x, expected)
```
Comment (nickrobinson251) [inline test/rulesets/Base/base.jl]: is there any other use-case for this? that would warrent it moving to `test/test_utils.jl`?

File: test/rulesets/Base/base.jl [hunk 2]
```diff
@@ -103,5 +103,22 @@ end
...
+            y, rule = frule(f, x)
+            @test y == expected
+            @test extern(rule(1))==0.0
```
Comment (nickrobinson251) [inline test/rulesets/Base/base.jl]: nitpick: 
```suggestion
            @test extern(rule(1)) == 0.0
```

File: test/rulesets/Base/base.jl [hunk 3]
```diff
@@ -103,5 +103,22 @@ end
...
+
+            y, rule = rrule(f, x)
+            @test y == expected
+            @test extern(rule(1))==0.0
```
Comment (nickrobinson251) [inline test/rulesets/Base/base.jl]:
```suggestion
            @test extern(rule(1)) == 0.0
```

File: test/rulesets/Base/base.jl (hunk 1 above)
Comment (oxinabox) [reply test/rulesets/Base/base.jl]: For now I am happy to leave it here.
We can move it later.

**Pull request merged**